### PR TITLE
Add workflows e2e test

### PR DESCRIPTION
**Pull request by nishkrishnan**
First stab at an e2e test that currently just checks to make sure that `UpdateCheckRuns` has been called enough times and that our stream handler has some output associated with 2 job ids.

This PR includes a lot of additional changes to make this work:
* Interfacing our github client
* Support config to auto apply
* Refactoring DI to allow us to inject a few custom things for testing (local cache, custom go-getter, gh client)

### CODE REVIEW

File: server/neptune/workflows/internal/deploy/request/root.go [hunk 1]
```diff
@@ -14,14 +14,17 @@ const (
...
-	TfVersion   string
-	PlanMode    PlanMode
-	Trigger     Trigger
+	Name             string
+	Apply            Job
+	Plan             Job
+	RepoRelPath      string
+	TfVersion        string
+	PlanMode         PlanMode
+	PlanApprovalType string
```
Comment (samrabelachew) [inline server/neptune/workflows/internal/deploy/request/root.go:26]: Can you set this field on the gateway side when we signal to start the workflows?

**Review by samrabelachew — COMMENTED** [review]
looks great

File: server/neptune/workflows/internal/deploy/request/root.go (hunk 1 above)
Comment (nishkrishnan) [reply server/neptune/workflows/internal/deploy/request/root.go:26]: Why?

File: server/neptune/workflows/internal/activities/github.go [hunk 2]
```diff
@@ -28,9 +47,10 @@ const (
 const deploymentsDirName = "deployments"
 
 type githubActivities struct {
-	ClientCreator githubapp.ClientCreator
-	DataDir       string
-	LinkBuilder   LinkBuilder
+	Client      githubClient
+	DataDir     string
+	LinkBuilder LinkBuilder
+	Getter      gogetter
```
[the remark this answers is not in the document]
Comment (nishkrishnan) [reply server/neptune/workflows/internal/activities/github.go:53]: yep

[the remark this answers is not in the document]
Comment (nishkrishnan) [reply server/neptune/workflows/internal/activities/github.go:53]: It could just as well be an interface, I have no opinion.

**Review by samrabelachew — APPROVED** [review]

**Review by samrabelachew — APPROVED** [review]

**Pull request merged**